tests/unit/CoreQuoteTest.php

```diff
@@ -0,0 +1,244 @@
+				'lock'            => null,
+				'style'           => null,
+				'textColor'       => null,
+				'value'           => '<p>This is a sample quote block content.</p>',
+			],
+			$block['attributes'],
+		);
+	}
+
+	/**
+	 * Test case for retrieving core quote block untested attributes.
+	 *
+	 * Covers : 'anchor', 'backgroundColor', 'fontFamily', 'fontSize', 'gradient', 'lock', 'style', 'textColor' and 'value'.
+	 */
+	public function test_retrieve_core_quote_additional_attributes(): void {
+		$block_content = '
+			<!-- wp:quote {"lock":{"move":true,"remove":true},"fontFamily":"body","fontSize":"small","backgroundColor":"pale-cyan-blue","style":{"elements":{"heading":{"color":{"text":"var:preset|color|vivid-cyan-blue","background":"var:preset|color|cyan-bluish-gray"}}}},"textColor":"vivid-red","gradient":"pale-ocean"} -->
+			<blockquote class="wp-block-quote" id="test-anchor"><!-- wp:paragraph -->
+			<p>Sample Quote</p>
+			<!-- /wp:paragraph --><cite>Citation</cite></blockquote>
+			<!-- /wp:quote -->
+		';
+
+		// Update the post content with the block content.
+		wp_update_post(
+			[
+				'ID'           => $this->post_id,
+				'post_content' => $block_content,
+			]
+		);
+
+		$query     = $this->query();
+		$variables = [
+			'id' => $this->post_id,
+		];
+
+		$actual = graphql( compact( 'query', 'variables' ) );
+
+		$this->assertArrayNotHasKey( 'errors', $actual, 'There should not be any errors' );
+		$this->assertArrayHasKey( 'data', $actual, 'The data key should be present' );
+		$this->assertArrayHasKey( 'post', $actual['data'], 'The post key should be present' );
+
+		$this->assertEquals( $this->post_id, $actual['data']['post']['databaseId'], 'The post ID should match' );
+
+		// Verify the block data.
+		$block = $actual['data']['post']['editorBlocks'][0];
+
+		$this->assertNotEmpty( $block['apiVersion'], 'The apiVersion should be present' );
+		$this->assertEquals( 'text', $block['blockEditorCategoryName'], 'The blockEditorCategoryName should be text' );
+		$this->assertNotEmpty( $block['clientId'], 'The clientId should be present' );
+
+		$this->assertNotEmpty( $block['innerBlocks'], 'There should be no inner blocks' );
+		$this->assertEquals( 1, count( $block['innerBlocks'] ), 'There should be only one inner block' );
+		$this->assertEquals( 'core/paragraph', $block['innerBlocks'][0]['name'], 'The inner block name should be core/paragraph' );
+
+		$this->assertEquals( 'core/quote', $block['name'], 'The block name should be core/quote' );
+		$this->assertEmpty( $block['parentClientId'], 'There should be no parentClientId' );
+		$this->assertNotEmpty( $block['renderedHtml'], 'The renderedHtml should be present' );
+
+		// Verify the attributes.
+		$this->assertEquals(
+			[
+				'anchor'          => 'test-anchor', // Previously untested.
+				'backgroundColor' => 'pale-cyan-blue', // Previously untested.
+				'citation'        => 'Citation',
+				'className'       => null,
+				'cssClassName'    => 'wp-block-quote',
```

Comment:
Unset this and leave a note that it's being tested above.

Reply:
Done ✅

Reply:
My stupid IDE. this should only have been highlighted this for the `cssClassName` that we're handling specially for WP6.4+ compatibility.